dynamiqs/utils/tensor_types.py

```diff
@@ -1,202 +1,81 @@
 from __future__ import annotations
 
-from typing import Union, get_args
-
+import jax.numpy as jnp
 import numpy as np
 import torch
+from jax.typing import ArrayLike
 from qutip import Qobj
-from torch import Tensor
 
-from .._utils import hdim, obj_type_str
-from .utils import isbra, isket, isop
+from .utils import _hdim, isbra, isket, isop
 
-__all__ = ['to_tensor', 'to_numpy', 'to_qutip', 'from_qutip']
+__all__ = ['to_qutip']
 
-Number = Union[int, float, complex]
+# TODO: remove (keep name to avoid ImportError while transitioning from PyTorch to JAX)
+to_device = None
+to_tensor = None
+to_numpy = None
+Number = None
 
-# type for objects convertible to a torch.Tensor using `to_tensor`
-ArrayLike = Union[tuple, list, np.ndarray, Tensor, Qobj]
 
-# data type conversion dictionaries
-DTYPE_TO_REAL = {torch.complex64: torch.float32, torch.complex128: torch.float64}
-DTYPE_TO_COMPLEX = {torch.float32: torch.complex64, torch.float64: torch.complex128}
+def _get_default_dtype() -> jnp.float32 | jnp.float64:
+    default_dtype = jnp.array(0.0).dtype
+    return jnp.float64 if default_dtype == jnp.float64 else jnp.float32
```

Comment:
```suggestion
    return jnp.array(0.0).dtype
```

Reply:
It does not return `jnp.float32` or `jnp.float64` but a weird object 😅 So then when you do `dtype is jnp.float32` it returns `False`. Maybe we should extend the typing of our dtype to match any possible dtype (e.g. in JAX you can do `dtype='complex32'`). Wdyt?

Reply:
Quite strange, but I guess we should do this:
```python
dtype_array = jnp.array(0.0, dtype=jnp.float32).dtype
dtype_numpy = np.dtype(np.float32)
print(dtype_array is dtype_numpy)
# True
```